src/utils/string/StringUtils.js

```diff
@@ -14,7 +14,8 @@ import {getAnonymousUserId} from '../users';
 import {getUserId} from '../token';
 
 const NO_POLL_UUID = '00000000-0000-0000-0000-000000000000';
-
+const urlRegex =
+  /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(\:\d+)?(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `urlRegex` seems to be a comprehensive regular expression for URL validation. However, it's quite complex and may not be easily understandable or maintainable. Consider adding a comment explaining what it does or breaking it down into smaller, more manageable parts.

```diff
+ // This regex validates URLs with or without http(s), domain names with 2 or more segments separated by dots,
+ // IP addresses, optional port numbers, paths, queries, and fragments.
const urlRegex =
  /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(\:\d+)?(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
it conflict with sonarcloud if I use this regex it will be reject it